homeassistant/components/ads/sensor.py

```diff
@@ -52,27 +67,51 @@ def setup_platform(
 ) -> None:
     """Set up an ADS sensor device."""
     ads_hub = hass.data.get(ads.DATA_ADS)
-
+    if ads_hub is None:
+        raise ValueError("ADS Hub is not initialized.")
```

Comment:
I wonder => can this really be missing from `hass.data`?
I would expect that if it is not present, then `setup_platform` is never reached.

You could maybe simply replace with `ads_hub: ads.AdsHub = hass.data[ads.DATA_ADS]`

https://github.com/home-assistant/core/blob/187a38c91fea8f8bda798e15cae53bee2ee48066/homeassistant/components/ads/__init__.py#L136-L147